Services/Database/interfaces/interface.ilDBInterface.php

```diff
@@ -293,4 +295,12 @@ public function buildAtomQuery(): ilAtomQuery;
     public function groupConcat(string $a_field_name, string $a_seperator = ",", ?string $a_order = null): string;
 
     public function cast(string $a_field_name, string $a_dest_type): string;
+
+    public function addForeignKey(string $foreign_key_name, string $field_name, string $table_name, string $reference_field_name, string $reference_table, ?string $on_update = null, ?string $on_delete = null): bool;
```

Comment:
We must accept a collection of fields/columns here which can be referenced, because a primary key in the referenced table can be a combined key.

Reply:
Fixed.